internal/manifests/collector/parser/receiver/receiver.go

```diff
@@ -0,0 +1,186 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package receiver
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/manifests/collector/parser"
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/naming"
+)
+
+// registry holds a record of all known receiver parsers.
+var registry = make(map[string]parser.Builder)
+
+// BuilderFor returns a parser builder for the given receiver name.
+func BuilderFor(name string) parser.Builder {
+	builder := registry[receiverType(name)]
+	if builder == nil {
+		builder = NewGenericReceiverParser
+	}
+
+	return builder
+}
+
+// For returns a new parser for the given receiver name + config.
+func For(logger logr.Logger, name string, config map[interface{}]interface{}) (parser.ComponentPortParser, error) {
+	builder := BuilderFor(name)
+	return builder(logger, name, config), nil
+}
+
+// Register adds a new parser builder to the list of known builders.
+func Register(name string, builder parser.Builder) {
+	registry[name] = builder
+}
+
+// IsRegistered checks whether a parser is registered with the given name.
+func IsRegistered(name string) bool {
+	_, ok := registry[name]
+	return ok
+}
+
+var (
+	endpointKey      = "endpoint"
+	listenAddressKey = "listen_address"
+	scraperReceivers = map[string]struct{}{
```

Comment:
nit: Can we narrow this down to what we actually support (e.g. prometheus)?

Reply:
updated to only supported receivers